### ISSUE TITLE
FloatingActionButton can't show/hide after call setFab()

### ISSUE THREAD

**Issue by jordenchang55** [Ashok-Varma/BottomNavigation#54]
I have 4 fragments and use BottomNavigation to switch between them.
In some fragments, I don't want to use fab, so I try to call <code>mFAB.hide()</code> to hide it.
No matter I call this method in activity or fragment, it just doesn't work, the button still there.

But if I remove the line, 
<code>mBottomNavigationBar.setFab(mFAB)</code>,
the button will show/hide as expected when switching between tabs.

**Comment by Ashok-Varma** [comment]
The Bug is due to animations. mFab.hide() uses animation to hide the fab. But at the same time since bottomBar state changed it is trying to animate fab cancelling previous animation so hide animation is stopped. i will try to fix it in upcoming release.
for now instead of 
```java 
mFAB.hide()
``` 
try to call
```java
mFAB.setVisibility
```

**Comment by SailFlorve** [comment]
Yeah I have the same case and met this bug, longing for fixing it!

**Comment by ahmadmust8** [comment]
it work if you write it like this 

```
override fun onTabSelected(position: Int) {
       
       if (position == 2){
            fab.visibility = View.GONE
        }else{
            fab.visibility = View.VISIBLE
        }
    }
